src/main/java/com/team2813/vision/MultiPhotonPoseEstimator.java

```diff
@@ -57,19 +75,43 @@ public void addToSim(
         });
   }
 
+  private record CameraConfig(Transform3d robotToCamera, Optional<String> description) {}
+
   private record CameraData(
-      PhotonCamera camera, PhotonPoseEstimator estimator, PhotonVisionPosePublisher publisher) {}
+      PhotonCamera camera,
+      PhotonPoseEstimator estimator,
+      Transform3d robotToCamera,
+      PhotonVisionPosePublisher publisher,
+      StructPublisher<Pose3d> cameraPosePublisher) {}
 
   private MultiPhotonPoseEstimator(Builder builder) {
-    for (Map.Entry<String, Transform3d> entry : builder.cameras.entrySet()) {
+    for (Map.Entry<String, CameraConfig> entry : builder.cameraConfigs.entrySet()) {
       String cameraName = entry.getKey();
       PhotonCamera camera = new PhotonCamera(builder.ntInstance, cameraName);
-      Transform3d robotToCamera = entry.getValue();
+      CameraConfig cameraConfig = entry.getValue();
       PhotonPoseEstimator estimator =
-          new PhotonPoseEstimator(builder.fieldTags, builder.poseStrategy, robotToCamera);
-      var publisher = new PhotonVisionPosePublisher(camera);
+          new PhotonPoseEstimator(
+              builder.fieldTags, builder.poseStrategy, cameraConfig.robotToCamera);
+      var estimatedPosePublisher = new PhotonVisionPosePublisher(camera);
+      NetworkTable table = getTableForCamera(camera);
+      var cameraPosePublisher = table.getStructTopic("cameraPose", Pose3d.struct).publish();
+
+      cameraDatas.add(
+          new CameraData(
+              camera,
+              estimator,
+              cameraConfig.robotToCamera,
+              estimatedPosePublisher,
+              cameraPosePublisher));
+
+      cameraConfig.description.ifPresent(
+          description -> table.getEntry("description").setString(description));
+    }
+  }
 
-      cameraDatas.add(new CameraData(camera, estimator, publisher));
+  public void setDrivePose(Pose2d pose) {
+    for (CameraData cameraData : cameraDatas) {
+      cameraData.cameraPosePublisher.set(new Pose3d(pose).plus(cameraData.robotToCamera));
     }
```

Comment:
the [`Pose3d#plus(Transform3d)`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Pose3d.html#plus(edu.wpi.first.math.geometry.Transform3d)) method doesn't modify the `Pose3d` it is called on, so we can move the new Pose3d outside of the loop (this isn't critical, and can be done later if we want to)
```suggestion
    Pose3d pose3d = new Pose3d(pose)
    for (CameraData cameraData : cameraDatas) {
      cameraData.cameraPosePublisher.set(pose3d.plus(cameraData.robotToCamera));
    }
```